src/nunavut/lang/c/templates/UnionType.j2

```diff
@@ -0,0 +1,10 @@
+{% extends "Header.j2" %}
+{%- block object -%}
+{% set composite_type = T -%}
+{% include '_composite_type.j2' %}
+{%- for field in composite_type.fields -%}
+{%- if field is not padding %}
+#define IS_{{ composite_type | full_reference_name }}_{{ field.name | macrofy(partial_token=True) }}(TAGGED_UNION_PTR) ((TAGGED_UNION_PTR) && (TAGGED_UNION_PTR)->_tag_ == {{ loop.index0 }})
```

Comment:
There is also a question of type safety and MISRA compliance. High-integrity guidelines generally prohibit the usage of function-like macros. Regardless of that, the macro-based design is prone to undetected type errors if the user feeds the pointer to a wrong union to the macro resulting in hard-to-detect runtime failures. Should we not use inline functions instead?

Reply:
inline functions it is then.